lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm

```diff
@@ -20,57 +20,69 @@ use Mojo::File;
 use POSIX 'strftime';
 
 sub index {
-    my $self   = shift;
+    my ($self, $obs_project, $folders) = @_;
     my $helper = $self->obs_rsync;
     my %folder_info_by_name;
-    my $folders = Mojo::File->new($helper->home)->list({dir => 1})->grep(sub { -d $_ })->map('basename')
-      ->grep(qr/^^(?!t$|profiles$|script$|xml$)/)->to_array;
+    $folders
+      = Mojo::File->new($helper->home, $obs_project // "")->list({dir => 1})->grep(sub { -d $_ })->map('basename')
+      ->grep(qr/^^(?!t$|profiles$|script$|xml$)/)->to_array
```

Comment:
What is the `^^` meant to do there? Why do you use a zero-width negative lookahead assertion?

Reply:
I'm surprised because that makes the regex very inefficient and it looks like something simple like `/^(?:t|profiles|script|xml)$/` would work just as well.

Reply:
Isn't the regex supposed to filter out those directories, and that's why it's a negative look-ahead? `/^(?:t|profiles|script|xml)$/` would be the opposite and match only those directories.

Reply:
Oh, yea of course, it has to be a `grep` block then.
```perl
@$folders = grep { !/^(?:t|profiles|script|xml)$/ } @$folders;
```

Reply:
Yeah, `^^` is a typo - sorry that it hurts so much. I don't see what is `very inefficient` and how replacing lookahead with `grep` block is a `must`

Reply:
Lookaheads/lookbehinds are much more costly operations for the regex engine, so it's customary to avoid them and use more efficient constructs when possible.